<?xml version="1.0"?>
<robot name="cepheus">
 
  <link name="cepheus">
    <visual>
      <material name="silver"> <color rgba="0.5 0.5 0.5 1"/> </material>
      <geometry>
        <!--mesh filename="/home/cepheus/catkin_ws/src/cepheus_robot/meshes/SpaceRobot.STL" /-->
        <cylinder length="0.0037" radius="0.0015"/>
      </geometry>
      <origin xyz="0 0 0.245" rpy="0 0 0"/>
    </visual>
    <visual>
      <material name="silver"> <color rgba="0.2 0.2 0.2 1"/> </material>
      <geometry>
        <box size="0.254 0.02 0.02 "/>
      </geometry>
      <origin xyz="0.268 0.0 0.1698" rpy="0 0 0"/>
    </visual>
  </link>

  <link name="reaction_wheel">
    <visual>
      <material name="silver"> <color rgba="0.5 0.5 0.5 1"/> </material>
      <geometry>
        <cylinder length="0.03" radius="0.055"/>
      </geometry>
      <origin xyz="0 0 0.015" rpy="0 0 0"/>
    </visual>
  </link>

  <link name="camera">
    <visual>
      <material name="camera"> <color rgba="0.2 0.2 0.2 1"/> </material>
      <geometry>
        <box size="0.018 0.0254 0.01 "/>
      </geometry>
      <origin xyz="0 0.005 -0.005" rpy="0 0 0"/>
    </visual>
    <visual>
      <material name="camera"> </material>
      <geometry>
        <cylinder length="0.003" radius="0.0055"/>
      </geometry>
      <origin xyz="0 0 0.0015" rpy="0 0 0"/>
    </visual>
  </link>

  <link name="probe_ft_sensor">
    <visual>
      <material name="botasys"> <color rgba="0.1 0.1 0.1 1"/> </material>
      <geometry>
        <cylinder length="0.026" radius="0.04"/>
      </geometry>
      <origin xyz="0 0 -0.013" rpy="0 0 0"/>
    </visual>
  </link>

  <link name="probe">
    <visual>
      <material name="silver"> <color rgba="0.5 0.5 0.5 1"/> </material>
      <geometry>
        <cylinder length="0.09955" radius="0.002"/>
      </geometry>
      <origin xyz="-0.049775 0 0 " rpy="0 1.570796327 0"/>
    </visual>
    <visual>
      <material name="silver"> </material>
      <geometry>
        <sphere radius="0.004"/>
      </geometry>
    </visual>
    <visual>
      <material name="silver"> </material>
      <geometry>
        <cylinder length="0.0016" radius="0.046"/>
      </geometry>
      <origin xyz="-0.07275 0 0 " rpy="0 1.570796327 0"/>
    </visual>
  </link>

  <!--link name="right_mouse">
    <visual>
      <geometry>
        <box size="0.09 .05 .03"/>
      </geometry>
      <origin xyz="0 0 0.015" rpy="0 0 0"/>
    </visual>
    <inertial>
      <origin xyz="0 0 0.015" rpy="0 0 0"/>
      <mass value="15"/>
      <inertia ixx="0.04" ixy="0.0" ixz="0.0" iyy="0.04" iyz="0.0" izz="0.02"/>
    </inertial>
  </link>

  <link name="left_mouse">
    <visual>
      <geometry>
        <box size="0.09 .05 .03"/>
      </geometry>
      <origin xyz="0 0 0.015" rpy="0 0 0"/>
    </visual>
    <inertial>
      <origin xyz="0 0 0.015" rpy="0 0 0"/>
      <mass value="15"/>
      <inertia ixx="0.04" ixy="0.0" ixz="0.0" iyy="0.04" iyz="0.0" izz="0.02"/>
    </inertial>
  </link-->

  <!--link name="T1">
    <visual>
      <material name="thruster"> <color rgba="0 0 0.5 1"/> </material>
      <geometry>
        <cylinder length="0.05" radius="0.01"/>
      </geometry>
      <origin xyz="0 0 0" rpy="0 1.57 0"/>
    </visual>
    <inertial>
      <origin xyz="0 0 0" rpy="0 0 0"/>
      <mass value="1"/>
      <inertia ixx="0.004" ixy="0.0" ixz="0.0" iyy="0.004" iyz="0.0" izz="0.002"/>
    </inertial>
  </link>

  <link name="T2">
    <visual>
      <material name="thruster"> <color rgba="0 0 0.5 1"/> </material>
      <geometry>
        <cylinder length="0.05" radius="0.01"/>
      </geometry>
      <origin xyz="0 0 0" rpy="0 1.57 0"/>
    </visual>
    <inertial>
      <origin xyz="0 0 0" rpy="0 0 0"/>
      <mass value="1"/>
      <inertia ixx="0.004" ixy="0.0" ixz="0.0" iyy="0.004" iyz="0.0" izz="0.002"/>
    </inertial>
  </link>

  <link name="T3">
    <visual>
      <material name="thruster"> <color rgba="0 0 0.5 1"/> </material>
      <geometry>
        <cylinder length="0.05" radius="0.01"/>
      </geometry>
      <origin xyz="0 0 0" rpy="0 1.57 0"/>
    </visual>
    <inertial>
      <origin xyz="0 0 0" rpy="0 0 0"/>
      <mass value="1"/>
      <inertia ixx="0.004" ixy="0.0" ixz="0.0" iyy="0.004" iyz="0.0" izz="0.002"/>
    </inertial>
  </link-->



  <joint name="reaction_wheel_joint" type="continuous">
    <parent link="cepheus"/>
    <child link="reaction_wheel"/>
    <origin xyz="0.07 0 0" rpy="0 0 0"/>
    <axis xyz="0 0 1" /> 
    <!--limit effort="10" velocity="200.5"/-->
  </joint>

  <joint name="camera_joint" type="fixed">
    <parent link="cepheus"/>
    <child link="camera"/>
    <origin xyz="0.44625 0.014 0.1698" rpy="0 1.570796327 0"/>
  </joint>

  <joint name="botasys_joint" type="fixed">
    <parent link="cepheus"/>
    <child link="probe_ft_sensor"/>
    <origin xyz="0.41945 0.0 0.1698" rpy="1.570796327 0 1.570796327"/>
  </joint>

  <joint name="probe_joint" type="fixed">
    <parent link="cepheus"/>
    <child link="probe"/>
    <origin xyz="0.519 0.0 0.1698" rpy="0 0 0"/>
  </joint>

  <!--joint name="r_odom" type="fixed">
    <parent link="cepheus"/>
    <child link="right_mouse"/>
    <origin xyz="0 -0.1 0" rpy="0 0 0"/>
  </joint>

  <joint name="l_odom" type="fixed">
    <parent link="cepheus"/>
    <child link="left_mouse"/>
    <origin xyz="0 0.1 0" rpy="0 0 0"/>
  </joint>



  <joint name="j_T1" type="fixed">
    <parent link="cepheus"/>
    <child link="T1"/>
    <origin xyz="-0.085 0.147224319 0.23" rpy="0 0 0.523598776"/>
  </joint>

  <joint name="j_T2" type="fixed">
    <parent link="cepheus"/>
    <child link="T2"/>
    <origin xyz="0.17 0 0.23" rpy="0 0 -1.570796327"/>
  </joint>

  <joint name="j_T3" type="fixed">
    <parent link="cepheus"/>
    <child link="T3"/>
    <origin xyz="-0.085 -0.147224319 0.23" rpy="0 0 -0.523598776"/>
  </joint-->
  
</robot>
